test/include/dlaf_test/util_matrix.h

```diff
@@ -0,0 +1,150 @@
+}
+
+/// @brief Sets the element of the matrix.
+/// The (i, j)-element of the matrix is set to el({i, j}).
+/// @pre el argument is an index of type const GlobalElementIndex&.
+/// @pre el return type should be T.
+template <class T, class Func>
+void set(Matrix<T, Device::CPU>& mat, Func el) {
+  for (SizeType tile_j = 0; tile_j < mat.nrTiles().cols(); ++tile_j) {
+    for (SizeType tile_i = 0; tile_i < mat.nrTiles().rows(); ++tile_i) {
+      auto tile = mat(LocalTileIndex(tile_i, tile_j)).get();
+      for (SizeType jj = 0; jj < tile.size().cols(); ++jj) {
+        SizeType j = tile_j * mat.blockSize().cols() + jj;
+        for (SizeType ii = 0; ii < tile.size().rows(); ++ii) {
+          SizeType i = tile_i * mat.blockSize().rows() + ii;
+          tile({ii, jj}) = el({i, j});
+        }
+      }
+    }
+  }
+}
+
+/// @brief Returns a col-major ordered vector with the future to the matrix tiles.
+template <class T, Device device>
+std::vector<hpx::future<Tile<T, device>>> getFutures(Matrix<T, device>& mat) {
+  std::vector<hpx::future<Tile<T, device>>> result;
+  result.reserve(util::size_t::mul(mat.nrTiles().rows(), mat.nrTiles().cols()));
+
+  for (SizeType j = 0; j < mat.nrTiles().cols(); ++j) {
+    for (SizeType i = 0; i < mat.nrTiles().rows(); ++i) {
+      result.emplace_back(std::move(mat(LocalTileIndex(i, j))));
+      EXPECT_TRUE(result.back().valid());
+    }
+  }
+  return result;
+}
+
+/// @brief Returns a col-major ordered vector with the read-only shared-future to the matrix tiles.
+template <class T, Device device>
+std::vector<hpx::shared_future<Tile<const T, device>>> getSharedFutures(Matrix<T, device>& mat) {
+  std::vector<hpx::shared_future<Tile<const T, device>>> result;
+  result.reserve(util::size_t::mul(mat.nrTiles().rows(), mat.nrTiles().cols()));
+
+  for (SizeType j = 0; j < mat.nrTiles().cols(); ++j) {
+    for (SizeType i = 0; i < mat.nrTiles().rows(); ++i) {
+      result.emplace_back(mat.read(LocalTileIndex(i, j)));
+      EXPECT_TRUE(result.back().valid());
+    }
+  }
+
+  return result;
+}
+
+/// @brief Checks the elements of the matrix.
+/// comp(el({i, j}), (i, j)-element) is used to compare the elements.
+/// err_message(el({i, j}), (i, j)-element) is printed for the first element which does not fulfill the comparison.
+/// @pre el argument is an index of type const GlobalElementIndex&.
+/// @pre comp should have two arguments and return true if the comparison is fulfilled and false otherwise.
+/// @pre err_message should have two arguments and return a string.
+/// @pre el return type should be the same as the type of the first argument of comp adn of err_message.
+/// @pre The second argument of comp should be either T, T& or const T&.
+/// @pre The second argument of err_message should be either T, T& or const T&.
+namespace internal {
+template <class T, class Func1, class Func2, class Func3>
+void check(Matrix<T, Device::CPU>& mat, Func1 el, Func2 comp, Func3 err_message, const char* file,
```

Comment:
This function do not assert. As written in the doc it checks if the elements fulfill the comparison and add a failure when it finds the first element that doesn't.
The function and macros which re supposed to be used by users (this is an internal detail) have the self explaining name.

Reply:
I said it was an assert-ish thing because, looking at the code I see, as you said, that you want to stop at the first failed comparison. I would have made it clear from the call that this is going to assert and stop the test.

However, it is ok for me. 👌🏼